queries.sql

```diff
@@ -23,3 +23,51 @@ SELECT * FROM animals WHERE name != 'Gabumon';
 
 /*Animals weighing between 10.4kg and 17.3kg*/
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+
+-- To update the animals table inside a transaction and set the species column to unspecified
+BEGIN TRANSACTION;
+UPDATE animals SET species = 'unspecified';
+SELECT * FROM animals WHERE species = 'unspecified';
+ROLLBACK TRANSACTION;
+
+-- Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
+-- Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
+BEGIN TRANSACTION;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL OR species = '';
+COMMIT TRANSACTION;
+
+-- delete wholw table and rollback
+BEGIN TRANSACTION;
+DELETE FROM animals;
+ROLLBACK TRANSACTION;
+
+-- Delete all animals born after Jan 1st, 2022.
+-- Create a savepoint for the transaction.
+-- Update all animals' weight to be their weight multiplied by -1.
+-- Rollback to the savepoint
+-- Update all animals' weights that are negative to be their weight multiplied by -1.
+-- Commit transaction
+BEGIN TRANSACTION;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT savepoint;
+UPDATE animals SET weight = weight * -1;
+ROLLBACK TO SAVEPOINT savepoint;
+UPDATE animals SET weight = weight * -1 WHERE weight < 0;
+COMMIT TRANSACTION;
+
+-- How many animals are there?
+SELECT COUNT(*) AS num_animals FROM animals;
+
+-- How many animals have never tried to escape?
+SELECT COUNT(*) AS num_never_escaped FROM animals WHERE escape_attempts = 0;
+
+-- What is the average weight of animals?
+SELECT AVG(weight_kg) AS avg_weight FROM animals;
+-- Who escapes the most, neutered or not neutered animals?
+SELECT neutered, SUM(escape_attempts) FROM animals GROUP BY neutered;
```

Comment:
- [ ] 👋 Hey there! Instead of returning the sum, why not opt for a more insightful approach by returning the maximum value of escape attempts? 📈🔝 This way, we can identify the animal, whether neutered or not, that has made the most escape attempts. 🐾🚀 It provides a clearer and more informative perspective on the data. What do you think of this idea? 🤔💡

```suggestion
SELECT neutered, MAX(escape_attempts) FROM animals
  GROUP BY neutered;
```